### DISCUSSION TITLE
Download statistic in releases

### DISCUSSION THREAD

**Discussion by awvwgk** [fortran-lang/fpm#620] [General]
One possibility to get an idea over the usage of fpm is to search for fpm projects as we did for the FortranCon and PackagingCon:

![fpm-projects](https://user-images.githubusercontent.com/28669218/142780536-704f3c79-f16b-457f-9fe3-c97d13f1f03f.png)

Always exciting to see long-running Fortran projects adopting fpm. But of course not every fpm project ends up on GitHub.

Another statistic we could look at are the downloads we are getting on our releases: ![GitHub all releases](https://img.shields.io/github/downloads/fortran-lang/fpm/total)
I looked a bit around at shields.io and we can indeed create a download statistic for our releases.

**Comment by ivan-pi** [comment]
Is the shield you show already linked to fpm? What is FC (with C inverted), I presume FortranCon?

**Reply by awvwgk** [reply]
Above I just used `![GitHub all releases](https://img.shields.io/github/downloads/fortran-lang/fpm/total)`

**Comment by urbanjost** [comment]
Downloads are interesting of themselves, but could be heavily skewed by things like CI/CD builds and cloud usage; I think the more critical value is the number of packages built; and the number of downloads would not even measure builds from single-source builds that can easily be built on an isolated system.  Can you tell how many pulls are from duplicate IP addresses and how many are from likely CI/CD sites like github?  For the future the public packages will be the most important to the Fortran eco-system.  I think we really need to work on a very simple-to-use registry.  

I still like to see the download numbers,  and things like the different number of sites and the amount of code in each project (several are just tests) and the number of downloads of the packages are all very interesting, as well as trends in the numbers that can help convince people of the relevance of fpm and encourage/discourage the developers as well, but they can also be very misleading; so the total number of downloads is not too important to me but the trend probably is.
